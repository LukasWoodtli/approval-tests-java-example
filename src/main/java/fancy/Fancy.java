package fancy;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Fancy {
    private final static Logger logger = LoggerFactory.getLogger("approval_test");

    public String hello() {
        String msg = "Hello";
        logger.debug("Message is: {}", msg);
        return msg;
    }
    public String hello(String who) {
        String msg = String.format("%s, %s", hello(), who);
        logger.debug("Complex message is: {}", msg);
        return msg;
    }
}
