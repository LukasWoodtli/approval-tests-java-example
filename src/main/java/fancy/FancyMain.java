package fancy;

public class FancyMain {
    public static void main(String[] args) {
        Fancy fancy = new Fancy();
        System.out.println(fancy.hello("World"));
    }
}
