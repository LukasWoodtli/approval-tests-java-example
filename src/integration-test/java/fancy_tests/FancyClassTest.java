package fancy_tests;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import fancy.Fancy;
import org.approvaltests.Approvals;
import org.junit.*;
import test_helper.LogAppenderTestRule;

import java.util.Arrays;


public class FancyClassTest
{
    @ClassRule
    public static LogAppenderTestRule appender = new LogAppenderTestRule("approval_test");

    @Test
    public void testList() throws Exception
    {
        String[] names = {"Llewellyn", "James", "Dan", "Jason", "Katrina"};
        Arrays.sort(names);
        Approvals.verifyAll("", names);
    }

    @Test
    public void testListToJson() throws Exception
    {
        String[] names = {"Llewellyn", "James", "Dan", "Jason", "Katrina"};
        Arrays.sort(names);

        Gson gson = new GsonBuilder().setPrettyPrinting().create();
        Approvals.verify(gson.toJson(names));
    }

    @Test
    public void testLogs() throws Exception
    {
        Fancy fancy = new Fancy();
        String greeting = fancy.hello("you");
        String logOutput = appender.getOutput();
        System.out.println("log output: " + logOutput);
        Approvals.verify(logOutput);
        Assert.assertEquals("Hello, you", greeting);
    }
}