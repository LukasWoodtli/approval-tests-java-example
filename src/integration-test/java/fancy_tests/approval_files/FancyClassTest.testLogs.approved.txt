approval_test DEBUG Message is: Hello
approval_test DEBUG Complex message is: Hello, you
